src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -65,6 +59,14 @@
 
     private Stage currentState = Stage.NA;
 
+    public ContextIndexSearcher(SearchContext searchContext, Engine.Searcher searcher, DirectoryReader filter) throws IOException {
+        super(filter);
+        // no push
```

Comment:
it is more of a question, here a new IndexSearcher is created instead of using the IndexSearcher from the provided Engine.Searcher. I'm not sure if this has bad consequences I didn't think about. I did this because I couldn't think of a different way to use the filtered directory reader in case an alias is used that has `fields` configured.